lstchain/scripts/lstchain_mc_dl2_to_irf.py

```diff
@@ -0,0 +1,170 @@
+    # calculate theta / distance between reco and assumed source position
+    mc_gamma["events"]["theta"] = calculate_theta(
+                    mc_gamma["events"],
+                    assumed_source_az=mc_gamma["events"]["true_az"],
+                    assumed_source_alt=mc_gamma["events"]["true_alt"],
+                    )
+    log.info(mc_gamma["simulation_info"])
+
+    # Apply selection cuts
+    gammas = mc_gamma["events"]
+
+    gh_cut = cuts["fixed_cuts"]["gh_score"][0]
+    log.info(f"Using fixed G/H cut of {gh_cut} to calculate theta cuts")
+
+    gammas = filter_events(gammas, cuts["events_filters"])
+
+    # Filtering the tels needed to use with the real data
+    # Add MAGIC tels when need be
+    tel_ids = cuts["LST_tels"]["tel_list"]
+    for i in tel_ids:
+        gammas["selected_tels"] = gammas["tel_id"] == i
+
+    gammas["selected_gh"] = gammas["gh_score"] > gh_cut
+    # irf_type = True for point like IRFs, False for Full Enclosure IRFs
+    if args.point_like:
+        gammas["selected_theta"] = gammas["theta"] < u.Quantity(**cuts["fixed_cuts"]["theta_cut"])
+        gammas["selected_fov"] = gammas["true_source_fov_offset"] < u.Quantity(**cuts["fixed_cuts"]["source_fov_offset"])
+        # Combining selection cuts
+        gammas["selected"] = gammas["selected_theta"] & \
+                            gammas["selected_gh"] & \
+                            gammas["selected_fov"] & \
+                            gammas["selected_tels"]
+    else:
+        gammas["selected"] = gammas["selected_gh"] & \
+                            gammas["selected_tels"]
+
+    # Binning of parameters used in IRFs
+    # 12.5 GeV - 51.28 TeV
+    true_energy_bins =  create_bins_per_decade(0.01 * u.TeV, 100 * u.TeV, 5.5)
+    # add_overflow_bins(***)[1:-1]
+    # The overflow binning added is not needed in the current script
+    reco_energy_bins = create_bins_per_decade(0.01 * u.TeV, 100 * u.TeV, 5.5)
+
+    # TODO: Generalize FoV offset binning
+    if mc_gamma["type"] == "point-like":
+        fov_offset_bins = [0.2, 0.6] * u.deg
+    else:
+        # temporary usage of bins as used in MAGIC
+        fov_offset_bins = [0,0.3,0.5,0.7,0.9,1.1] * u.deg
+    migration_bins = np.linspace(0.2, 5, 31)
+
+    # Write HDUs
+    hdus = [fits.PrimaryHDU(),]
+    with np.errstate(invalid='ignore', divide='ignore'):
+        if mc_gamma["type"] == "point-like":
+            effective_area = effective_area_per_energy(gammas[gammas["selected"]], mc_gamma["simulation_info"], true_energy_bins)
+        else:
+            effective_area = effective_area_per_energy_and_fov(gammas[gammas["selected"]], mc_gamma["simulation_info"], true_energy_bins, fov_offset_bins)
+    # Adding a dimension for FoV offset for effective area
+    hdus.append(create_aeff2d_hdu(effective_area[..., np.newaxis],true_energy_bins, fov_offset_bins, extname = "EFFECTIVE AREA"))
```

Comment:
Why do you need to add a dimension for the effective area? Is this related to a Gammapy bug? FYI, there is a discussion on this topic in the Slack of Gammapy dated of the 18-19 Jan 2021, with some ideas to deal with it.

Reply:
Yes. It was related to an issue with gammapy. I will update with the discussion on Slack and correct this.